contrib/build-functions.sh

```diff
@@ -0,0 +1,777 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+_missing_requirement() {
+    _error "ERROR: ${1:-} requires an argument."
+}
+
+# Error handling functions
+_error() {
+    printf '%s %s\n' "${1:-}" "${2:-}" >&2
+    return 1
+}
+
+_setup_environment() {
+    local install_dir=${_NIXL_INSTALL_DIR:-}
+
+    export LIBRARY_PATH=$LIBRARY_PATH:/usr/local/cuda/lib64
+
+    export LD_LIBRARY_PATH=${install_dir}/lib:${install_dir}/lib/x86_64-linux-gnu:${install_dir}/lib/x86_64-linux-gnu/plugins:/usr/local/lib:$LD_LIBRARY_PATH
+    export LD_LIBRARY_PATH=/usr/local/cuda/lib64:/usr/local/cuda/lib64/stubs:/usr/local/cuda/lib64:/usr/local/cuda-12.8/compat:$LD_LIBRARY_PATH
+    export LD_LIBRARY_PATH=/usr/local/cuda/compat/lib.real:$LD_LIBRARY_PATH
+
+    export CPATH=${install_dir}/include${CPATH:+:$CPATH}
+    export PATH=${install_dir}/bin${PATH:+:$PATH}
+    export PKG_CONFIG_PATH=${install_dir}/lib/pkgconfig${PKG_CONFIG_PATH:+:$PKG_CONFIG_PATH}
+    export NIXL_PLUGIN_DIR=${install_dir}/lib/x86_64-linux-gnu/plugins
+}
+
+# Show help message
+function show_help() {
+    set +x
+    local function_name=${1:-}
+    echo "usage: source contrib/build.env && <command> [options]"
+    echo "Examples: "
+    echo "  source contrib/build.env && install_build_dependencies"
+    echo "  source contrib/build.env && build_nixl --install-dir foo"
+    echo "  INSTALL_DIR=/foo source contrib/build.env && build_nixl"
+    echo "  source contrib/build.env && build_ucx --install-dir /bar"
+    echo "  UCX_INSTALL_DIR=/bar source contrib/build.env && build_ucx"
+    echo "  source contrib/build.env && install_test_dependencies"
+    echo "  source contrib/build.env && run_cpp_tests --install-dir /opt/nvidia/nixl"
+    echo "  source contrib/build.env && run_python_tests --install-dir /opt/nvidia/nixl"
+    echo ""
+    case $function_name in
+        build_nixl)
+            echo "Build the NIXL project"
+            echo ""
+            echo "Description:"
+            echo "  Builds the NIXL project using Meson build system. This function configures"
+            echo "  and compiles NIXL with the specified options and installs it to the target"
+            echo "  directory."
+            echo ""
+            echo "Options:"
+            echo "  --install-dir DIR           Installation directory for NIXL (default: $_NIXL_INSTALL_DIR)"
+            echo "  --ucx-install-dir DIR       UCX installation directory (default: $_UCX_INSTALL_DIR)"
+            echo "  --build-dir DIR             Build directory (default: $_NIXL_BUILD_DIR)"
+            echo "  --static-plugins LIST       Comma-separated list of plugins to build statically"
+            echo "  --enable-doxygen            Build API documentation using Doxygen (default: false)"
+            echo "  --no-enable-doxygen         Skip building documentation"
```

Comment:
One line in the help message is enough.
```suggestion
            echo "  --enable-doxygen            Build API documentation using Doxygen (default: disabled)"
```

Reply:
To avoid duplication of the same information.

Reply:
same